v1/bundle/store.go

```diff
@@ -668,44 +693,94 @@ func eraseData(ctx context.Context, store storage.Store, txn storage.Transaction
 	return nil
 }
 
-func erasePolicies(ctx context.Context, store storage.Store, txn storage.Transaction, parserOpts ast.ParserOptions, roots map[string]struct{}) (map[string]*ast.Module, error) {
+type moduleInfo struct {
+	RegoVersion ast.RegoVersion `json:"rego_version"`
+}
+
+func readModuleInfoFromStore(ctx context.Context, store storage.Store, txn storage.Transaction) (map[string]moduleInfo, error) {
+	versions := map[string]moduleInfo{}
+
+	value, err := read(ctx, store, txn, ModulesInfoBasePath)
+	if suppressNotFound(err) != nil {
+		return nil, err
+	}
+
+	if value != nil {
+		bs, err := json.Marshal(value)
+		if err != nil {
+			return nil, fmt.Errorf("corrupt rego version")
+		}
+
+		err = util.UnmarshalJSON(bs, &versions)
+		if err != nil {
+			return nil, fmt.Errorf("corrupt rego version")
+		}
```

Comment:
Why do we need to do this roundtripping, exactly? I'd suspect it's a little more annoying, but not harder, to do this via
```go
if m, ok := value.(map[string]any); ok {
  versions := make(map[string]moduleInfo, len(m))
  
  for k, v := range m {
    if m0, ok := v.(map[string]any); ok {
      if ver, ok := v["rego_version"]; ok {
        if vs, ok := ver.(int) { // float64?      
          versions[k] = moduleInfo{RegoVersion: fromInt(vs)}
        }
      }
    }
  }
  return versions, nil
}

return nil, fmt.Errorf("corrupt rego version")
```